Align if-match * and if-none-match * with RFC9110 (#1990)
oasis-tcs/odata-specs: odata-protocol/8 Header Fields.md

the case of an [upsert](#UpsertanEntity), if the addressed entity does
not exist the provided ETag value is considered not to match.

> An `If-Match` header with a value of `*` matches any current representation of the resource, and in a `PUT` or `PATCH` request

Reviewer: I find the following easier to grasp:
```suggestion
The precondition `If-Match: *` is fulfilled if a current representation of the resource exists, a `PUT` or `PATCH` request with that header
```

[`412 Precondition Failed`](#ResponseCode412PreconditionFailed) and MUST
ensure that no observable change occurs as a result of the request.

> An `If-None-Match` header with a value of `*` "matches" if there is no current representation of the resource, and in a `PUT` or `PATCH`

Reviewer: ```suggestion
The precondition `If-None-Match: *` is fulfilled if there is no current representation of the resource, a `PUT` or `PATCH` request with that header
```
